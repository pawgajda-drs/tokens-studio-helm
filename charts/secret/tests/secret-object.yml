suite: test secret

templates:
  - templates/secret.yaml

tests:
  - it: should render a secret with correct name, labels, annotations, and data
    set:
      external: foo
      secrets:
        app:
          labels:
            app: database
          annotations:
            app.kubernetes.io/source: "github"
          nameTemplate: '{{ .Values.external }}-secret'
          stringData:
            static: file
          stringDataTemplates:
            dynamic: '{{ title "app" }}'
          data:
            base64: aGVsbG8=
    asserts:
      - matchSnapshot: {}
  - it: should render stringData templates if no string data is defined
    set:
      external: foo
      secrets:
        app:
          labels:
            app: database
          annotations:
            app.kubernetes.io/source: "github"
          nameTemplate: '{{ .Values.external }}-secret'
          stringDataTemplates:
            dynamic: '{{ title "app" }}'
          data:
            base64: aGVsbG8=
    asserts:
      - matchSnapshot: {}
  - it: should render data templates 
    set:
      external: foo
      secrets:
        app:
          labels:
            app: database
          annotations:
            app.kubernetes.io/source: "github"
          nameTemplate: '{{ .Values.external }}-secret'
          dataTemplates:
            property1: '{{ .Values.external }}-value'
    asserts:
      - matchSnapshot: {}