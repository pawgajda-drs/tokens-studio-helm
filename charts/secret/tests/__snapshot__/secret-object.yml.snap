should render a secret with correct name, labels, annotations, and data:
  1: |
    apiVersion: v1
    data:
      base64: aGVsbG8=
    kind: Secret
    metadata:
      annotations:
        app.kubernetes.io/source: github
      labels:
        app: database
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: secrets
        helm.sh/chart: secrets-1.0.0
      name: foo-secret
    stringData:
      dynamic: App
      static: file
    type: Opaque
should render data templates:
  1: |
    apiVersion: v1
    data:
      property1: foo-value
    kind: Secret
    metadata:
      annotations:
        app.kubernetes.io/source: github
      labels:
        app: database
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: secrets
        helm.sh/chart: secrets-1.0.0
      name: foo-secret
    type: Opaque
should render stringData templates if no string data is defined:
  1: |
    apiVersion: v1
    data:
      base64: aGVsbG8=
    kind: Secret
    metadata:
      annotations:
        app.kubernetes.io/source: github
      labels:
        app: database
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: secrets
        helm.sh/chart: secrets-1.0.0
      name: foo-secret
    stringData:
      dynamic: App
    type: Opaque
